files/galaxy/tpv/tools.yml

```diff
@@ -1444,3 +1444,10 @@ tools:
   toolshed.g2.bx.psu.edu/repos/galaxyp/fragpipe/fragpipe/.*:
     cores: 8
     mem: 56
+    rules:
+      - if: |
```

Comment:
if think `- id` is missing, please look at the hifi example.

Reply:
I'm still getting `AttributeError: 'Tool' object has no attribute 'params_from_strings'` despite modeling hifiasm pretty closely. Any thoughts?

Reply:
@bgruening - I'm trying to dig into this further but I think I have `hifiasm` pretty well replicated, and I'm wondering if there's any piece I'm missing here?

Reply:
When I run `tpv dry-run` on the `toolshed.g2.bx.psu.edu/repos/galaxy-australia/hifiasm_meta/hifiasm_meta/0.3.1+galaxy0`, I get the same error. I don't know whether this rule was tested before or if something changed in the upstream codebase that handles the `tool` entity. I will have a further look and update you.

Reply:
Could it be [Job.params](https://github.com/galaxyproject/galaxy/blob/d3fb3fdf7b8ff898729b4ed3814a2a73fa77b872/lib/galaxy/model/__init__.py#L1457) instead of `Job.parameters`?
Just guessing here

Reply:
I tried dry running with older versions, back to 2.3.0 and the error remained the same

Reply:
The `hifiasm` seems to run when jobs are submitted via Galaxy (but the TPV dry-run still fails with the above error). The problem is probably with the `tpv dry-run` not getting access to the Galaxy apps `tool` object. I now remember seeing something similar that involves using the `tool` object of Galaxy. We can proceed with this rule since the `hifiasm` runs without issue.

However, when testing `fragpipe` , I see the following error:

```
[main] ERROR umich.ms.fileio.filetypes.bruker.Timsdata - JNA could not load native Bruker library
[main] ERROR umich.ms.fileio.filetypes.bruker.Timsdata - The ext folder with timsdata-2-21-0-4.dll (Win) or libtimsdata-2-21-0-4.so (Linux) is needed. If you are using an old ext folder, please upgrade.
```

Also, since this is a Java application, do we have to set any `env`? Like in [here ](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/master/files/galaxy/tpv/tools.yml#L1025-L1027)

Reply:
Thanks for looking into this @sanjaysrikakulam and @mira-miracoli!

That error is a known issue in the bioconda recipe and doesn't prevent the tool from running. We shouldn't need to set up the Java env either as the wrapper scripts do this already using the `GALAXY_MEMORY_MB` variable.

Reply:
The jobs fail as soon as they hit those errors. Maybe I am missing some parameters (I left everything to defaults)? Please test the tool and let us know if it works and we can proceed with this PR.

Reply:
Are these the automated tests as defined in the `<tests>` element that are running? These ran successfully in the most recent update here (https://github.com/galaxyproteomics/tools-galaxyp/pull/769), and we do have successful jobs on the EU site as well.

Reply:
Nope. I started a job through the UI. For the data, I clicked on `Generate Tour` and did not change any parameters in the tool's UI. The jobs failed with the above error. If your jobs run in the EU without errors, we can proceed with this PR.